docs/source/Getting-Started/Tutorials/Build-Enclave-App-Hello-World.rst

```diff
@@ -0,0 +1,257 @@
+Tutorial 2: Hello, World!
+======================================
+
+This tutorial explains how to build and run a simple "hello world" enclave.
+An enclave consists of an eapp and a runtime, but also needs the host that initializes and launches
+the enclave.
+Thus, each enclave source tree contains at least the host and eapp.
+
+Before jumping into the tutorial, please complete :doc:`Quick Start
+<../Running-Keystone-with-QEMU>`.
+
+
+Prerequisite
+------------------------------
+
+`musl-libc <https://www.musl-libc.org/>`_ is a lightweight standard library for replacing ``glibc``.
+The Eyrie runtime allows the enclave to be statically linked with musl, and supports a few standard
+functions such as ``printf``.
+
+Download source code in your work directory, and build musl with following commands.
+
+::
+
+	cd <your path>
+	git clone https://github.com/riscv/riscv-musl
+	cd riscv-musl
+	./configure --prefix=<install path (absolute path)> --disable-shared --enable-static
+	--host=riscv64-unknown-linux-gnu
+	make; make install
+
+``musl-gcc`` will be installed at ``<install path>``.
+
+Open ``sdk/examples/hello/eapp/Makefile`` and set ``CC`` to point ``musl-gcc``
+
+:: 
+
+	CC=<install path>/bin/musl-gcc
+
+
+Finally, set ``PATH`` to include RISC-V tools and ``KEYSTONE_SDK_DIR`` to point the
+absolute path to ``sdk`` directory.
+
+::
+
+	export PATH=$PATH:<path to RISC-V tools>
+	export KEYSTONE_SDK_DIR=<path to SDK>
+
+Let's take a look at the example provided in `Keystone SDK
+<https://github.com/keystone-enclave/keystone-sdk>`_.
+
+::
+
+	ls sdk/examples/hello
+
+You can find two directories and a script called ``vault.sh``
+
+vault.sh
```

Comment:
Vault description should maybe belong in its own doc?
   - When we want to land it there will be modifications to it as well
     that will need to be reflected to the doc. Some of these comments
     should be included in the script itself.

Reply:
Sure, but since other applications (e.g., keystone-demo, keystone-bench, etc) dont have vault.sh yet, I think it is okay to stay here until we have vault.sh for others and rewrite the entire "Keystone Application" section.